### PR TITLE
make Page implementations idempotent

### PR DESCRIPTION
**Pull request by ostumpf**

### CODE REVIEW

File: src/test/java/com/gooddata/collections/UriPageTest.java [hunk 1]
```diff
@@ -38,6 +38,8 @@ public void test() throws Exception {
         final UriComponentsBuilder builder = fromUriString("/this/is/{template}").query("other=false");
 
         uri.updateWithPageParams(builder);
+        uri.updateWithPageParams(builder);
```
Comment (martiner) [inline src/test/java/com/gooddata/collections/UriPageTest.java]: do we need this if we have testUpdateWithPageParamsIdempotency?

Comment (ostumpf) [inline src/test/java/com/gooddata/collections/UriPageTest.java]: Well `testUpdateWithPageParamsIdempotency` tests the `PageRequest` object while this tests `UriPage`... As far as I can see, these objects are two separate implementations of `Page`, so they deserve two separate tests.

Comment (martiner) [inline src/test/java/com/gooddata/collections/UriPageTest.java]: oh, i see. can you please make a bit explicit - create a new test method with a descriptive name

Comment (ostumpf) [inline src/test/java/com/gooddata/collections/UriPageTest.java]: done

**Pull request merged**